crates/cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -120,47 +132,35 @@ impl ParsedQObject {
         })
     }
 
-    fn parse_qml_metadata(
-        name: &Name,
-        attrs: &mut Vec<Attribute>,
-    ) -> Result<Option<QmlElementMetadata>> {
-        // Find if there is a qml_element attribute
-        if let Some(attr) = attribute_take_path(attrs, &["qml_element"]) {
+    fn parse_qml_metadata(name: &Name, attrs: &[Attribute]) -> Result<Option<QmlElementMetadata>> {
+        let attributes = parse_attributes(attrs, &Self::ALLOWED_ATTRS)?;
+        if let Some(attr) = attributes.get("qml_element") {
             // Extract the name of the qml_element from macro, else use the c++ name
             // This will use the name provided by cxx_name if that attr was present
-            let name = match attr.meta {
+            let name = match &attr.meta {
                 Meta::NameValue(name_value) => expr_to_string(&name_value.value)?,
                 _ => name.cxx_unqualified(),
             };
-
-            // Determine if this element is uncreatable
-            let uncreatable = attribute_take_path(attrs, &["qml_uncreatable"]).is_some();
-
-            // Determine if this element is a singleton
-            let singleton = attribute_take_path(attrs, &["qml_singleton"]).is_some();
-
+            let uncreatable = attributes.contains_key("qml_uncreatable");
+            let singleton = attributes.contains_key("qml_singleton");
             return Ok(Some(QmlElementMetadata {
                 name,
                 uncreatable,
                 singleton,
             }));
         }
-
         Ok(None)
     }
 
     fn parse_property_attributes(attrs: &mut Vec<Attribute>) -> Result<Vec<ParsedQProperty>> {
```

Comment:
wonder if this should be `&Vec<Attribute>`

Reply:
Drain needs a mutable reference which is used in parse_property_attributes

Reply:
See next comment, i don't think we should drain if we are not removing attributes